File: src/main/java/upbrella/be/rent/dto/response/ImprovementReportPageResponse.java
package upbrella.be.rent.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class ImprovementReportPageResponse {

    private List<ImprovementReportResponse> improvementReports;

    public static ImprovementReportPageResponse of(List<ImprovementReportResponse> improvementReports) {

        return ImprovementReportPageResponse.builder()
                .improvementReports(improvementReports)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/rent/dto/response/ImprovementReportPageResponse.kt
package upbrella.be.rent.dto.response

data class ImprovementReportPageResponse(
    val improvementReports: List<ImprovementReportResponse>
) {
    companion object {
        fun of(improvementReports: List<ImprovementReportResponse>): ImprovementReportPageResponse {
            return ImprovementReportPageResponse(improvementReports)
        }
    }
}
